### PR TITLE
Update Jenkins to support node selectors for agents.

### PR DESCRIPTION
**Pull request by kfox1111**

### CODE REVIEW

File: stable/jenkins/values.yaml [hunk 1]
```diff
@@ -87,6 +87,7 @@ Agent:
   # - type: Secret
   #   secretName: mysecret
   #   mountPath: /var/myapp/mysecret
+  NodeSelector: {}
```
Comment (lachie83) [inline stable/jenkins/values.yaml:92]: Could you please leave an example comments of how to use this?

File: stable/jenkins/templates/config.yaml [hunk 2]
```diff
@@ -36,7 +36,13 @@ data:
               <instanceCap>2147483647</instanceCap>
               <idleMinutes>0</idleMinutes>
               <label>{{ .Release.Name }}-{{ .Values.Agent.Component }}</label>
-              <nodeSelector></nodeSelector>
+              <nodeSelector>
+                {{- $local := dict "first" true }}
```
Comment (lachie83) [inline stable/jenkins/templates/config.yaml:40]: I have never seen this before in gotpl. Can you explain what you are doing here?

**Review by lachie83 — COMMENTED** [review]
Couple of quick comments. Please also bump the chart version

File: stable/jenkins/values.yaml (hunk 1 above)
Comment (kfox1111) [reply stable/jenkins/values.yaml:92]: I see a heck of a lot of documentation repeated over and over in each helm chart. They often get out of sync from each other. Should this belong in some kind of "how to use helm charts" document instead?

Comment (lachie83) [reply stable/jenkins/values.yaml:92]: Yeah I don't mind that idea at all. Let me get back to you on that. Thanks @kfox1111

File: stable/jenkins/templates/config.yaml (hunk 2 above)
Comment (kfox1111) [reply stable/jenkins/templates/config.yaml:40]: Yeah, gotl is horrible at this sort of thing. this is the only way I've figured out how to do this.... 

Gotl out of the box has no way of determining if you are in the first iteration (or last) iteration of a loop or not. But you need to know if you need to put comma's or some other thing on everything not the first run(or visa versa)

gotl's scoping rules also do not facilitate this sort of thing. You can't change the value of something already set. so you can't just 
{{ $first := true}}
range ....
{{ $first := false }}
end range

But, there is one caviot. Using the additional 'set' function, you can change the value of a key stored in a dict. This lets the $local.first value be overwritten inside the loop letting it keep state of whether its been through the loop once already or not, thus allowing commas to be added in the right places.

Yeah. its kind of a hack, but it works. haven't found a way to do it better.

**Pull request merged**